### PR TITLE
Merchant dashboard us37

### LINKED ISSUE #10: User Story 37, Admin can see a merchant's dashboard
As an admin user
When I visit the merchant index page ("/merchants")
And I click on a merchant's name,
Then my URI route should be ("/admin/merchants/6")
Then I see everything that merchant would see

### PR DESCRIPTION
**Pull request by priyapower**
closes #10 

**Contributors**
- Driver(s): Priya
- Navigator(s): Luke
  
**What was changed:**
- Add admin controller action updates, routes, and view pages for access to a merchant dashboard without hitting out authorization error (bypasses the 404 error that would occur if we tried to route it directly to merchant dashboard)

**Any tech notes (if applicable)**
- Refactor the dashboards for model methods and active record calls

### CODE REVIEW

File: app/controllers/admin/dashboard_controller.rb [hunk 1]
```diff
@@ -2,4 +2,37 @@ class Admin::DashboardController < Admin::BaseController
...
+      end
+    end
+    @pending_data = @pending_orders.zip(@pending_items.zip(@pending_item_orders))
+  end
+
+  def merchant_items
+    @merchant = Merchant.find(params[:merchant_id])
+    @items = Item.where(merchant_id:@merchant.id)
+  end
+
```
Comment (LHJE) [inline app/controllers/admin/dashboard_controller.rb:37]: We will be looking at this in the refactor, but I'm glad we were able to just use this same logic!

File: app/views/admin/dashboard/merchant_items.html.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<% @items.each do |item| %>
+  <%= item.name %>
+<% end  %>
```
Comment (LHJE) [inline app/views/admin/dashboard/merchant_items.html.erb:3]: Glad we added this for more robust testing.

File: app/views/merchants/index.html.erb [hunk 3]
```diff
@@ -3,7 +3,11 @@
 <section class = "grid-container">
   <% @merchants.each do |merchant|%>
     <section class = "grid-item">
-      <h2><%=link_to merchant.name, "/merchants/#{merchant.id}"%></h2>
+      <% if current_user.nil? || !current_user.role == "admin" %>
+        <h2><%=link_to merchant.name, "/merchants/#{merchant.id}"%></h2>
+      <% else current_user.role == "admin" %>
+        <h2><%=link_to merchant.name, "/admin/merchants/#{merchant.id}"%></h2>
+      <% end %>
```
Comment (LHJE) [inline app/views/merchants/index.html.erb:10]: Clever work around!

File: spec/features/merchant/show_spec.rb [hunk 4]
```diff
@@ -62,4 +64,37 @@
...
+      expect(page).to have_content(@order_1.created_at)
+      expect(page).to have_content(@item_order_1.quantity)
+      expect(page).to have_content(@item_order_1.quantity * @item_order_1.price)
+
+      expect(page).to have_link("Items for #{@bike_shop.name}")
+      click_link "Items for #{@bike_shop.name}"
+      expect(current_path).to eq("/admin/merchants/#{@bike_shop.id}/items")
+      expect(page).to have_content(@item_1.name)
+    end
+  end
```
Comment (LHJE) [inline spec/features/merchant/show_spec.rb:99]: Great job breaking this up into some more describe blocks.

**Review by LHJE — APPROVED** [review]

File: spec/features/merchant/show_spec.rb (hunk 4 above)
Comment (priyapower) [reply spec/features/merchant/show_spec.rb:99]: Woo! Thank you for teaching me that we CAN use these nested describes and before :eaches

**Pull request merged**